### PR TITLE
fix(node): vendor readable-stream from esm.sh

### PR DESCRIPTION
**Pull request by cjihrig**
~~Opening as a draft PR to see how the CI goes while I work on a script to automate the vendoring process.~~

### CODE REVIEW

**Review by bartlomieju — COMMENTED** [review]
Massive effort @cjihrig, thank you so much for looking into this. This looks mostly good to me and I'm waiting eagerly for fixes in `readable-stream` that will allow us to remove even more code. Could you add some README file on how to update `node/_stream.js` file?

**Review by bartlomieju — APPROVED** [review]
LGTM

File: node/_tools/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+This directory contains tooling for implementing Deno's Node.js compatibility
+layer.
+
+## Updating the Streams Implementation
+
+The Node.js streams implementation is based on the `readable-stream` module on
+npm. To update the code, run the following command:
+
+```
+deno run -A --unstable node/_tools/vendor_readable_stream.ts
```
Comment (kt3k) [inline node/_tools/README.md:12]: 👍

**Review by kt3k — APPROVED** [review]
LGTM. Great!

**Pull request merged**